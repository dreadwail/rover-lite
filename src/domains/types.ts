import { Action, AnyAction } from 'redux';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';

import { RoverSearchState } from '../state/types';

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export type GenericThunkAction<T, S extends RoverSearchState = RoverSearchState> = ThunkAction<
  T,
  S,
  {},
  AnyAction
>;
export type PromiseThunkAction<
  T,
  S extends RoverSearchState = RoverSearchState
> = GenericThunkAction<Promise<T>, S>;

export type MapDispatchToProps<
  TDispatchProps,
  TOwnProps = {},
  TState extends RoverSearchState = RoverSearchState,
  A extends Action = Action
> = <E>(dispatch: ThunkDispatch<TState, E, A>, ownProps: TOwnProps) => TDispatchProps;
