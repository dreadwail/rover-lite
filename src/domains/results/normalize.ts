import { Listing } from '../../state/results/types';

export interface RawDerivedData {
  readonly zip_code: string;
}

export interface RawResponsiveness {
  readonly photo_rate?: number;
  readonly photo_rate_desc: string;
  readonly percent?: number;
  readonly percent_desc: string;
  readonly time_short_desc: string;
}

export interface RawImage {
  readonly small: string;
}

export interface RawAttribute {
  readonly text: string;
  readonly key: string;
  readonly icon: string;
}

export interface RawPreferences {
  readonly dogs_preferences_attributes: RawAttribute[];
  readonly accepted_dogs_attributes: RawAttribute[];
  readonly dogs_experience_attributes: RawAttribute[];
  readonly attributes: RawAttribute[];
}

export interface RawProviderProfile {
  readonly hosting_service_preferences: RawPreferences;
  readonly traveling_service_preferences: RawPreferences;
  readonly attributes: RawAttribute[];
  readonly years_of_experience: number;
}

export interface RawBadge {
  readonly icon: string;
  readonly description: string;
  readonly title: string;
}

export interface RawPersonSummary {
  readonly responsiveness: RawResponsiveness;
  readonly first_name: string;
  readonly reviews_count: number;
  readonly profile_photo: RawImage;
  readonly ratings_average: string;
  readonly web_url: string;
  readonly images: RawImage[];
  readonly provider_profile: RawProviderProfile;
  readonly attributes: Record<string, RawAttribute>;
  readonly badges: RawBadge[];
}

export interface RawBadgeData {
  readonly content: string;
  readonly image_url: string;
  readonly show_on_search: boolean;
  readonly title: string;
}

export interface RawReview {
  readonly content: string;
  readonly dog_photo_url: string;
}

export interface RawResult {
  readonly rank: number;
  readonly person_summary: RawPersonSummary;
  readonly reviews_count: number;
  readonly neighborhood: string;
  readonly price_unit: string;
  readonly city: string;
  readonly zip: string;
  readonly title: string;
  readonly state: string;
  readonly contact_url: string;
  readonly person_name: string;
  readonly price: string;
  readonly review_text: string;
  readonly review_dog_photo_url: string;
  readonly badge_data: RawBadgeData[];
  readonly repeat_client_count: number;
  readonly reviews: RawReview[];
  readonly ratings_average: string;
  readonly currency_symbol: string;
}

export interface RawSearchResponse {
  readonly results: RawResult[];
}

const rawResultToListing = (rawResult: RawResult): Listing => ({
  thumbnailUrl: rawResult.person_summary.profile_photo.small,
  rank: rawResult.rank,
  name: rawResult.person_summary.first_name,
  badges: rawResult.badge_data.map(rawBadge => ({
    content: rawBadge.content,
    imageUrl: rawBadge.image_url,
    title: rawBadge.title,
  })),
  tagline: rawResult.title,
  neighborhood: rawResult.neighborhood,
  city: rawResult.city,
  state: rawResult.state,
  zip: rawResult.zip,
  repeatClientCount: rawResult.repeat_client_count,
  reviewCount: rawResult.reviews_count,
  ratingAverage: parseFloat(rawResult.ratings_average),
  featuredReview: rawResult.review_text,
  currencySymbol: rawResult.currency_symbol,
  price: parseFloat(rawResult.price),
  priceUnit: rawResult.price_unit,
  webUrl: rawResult.person_summary.web_url,
});

export const normalize = (rawResponse: RawSearchResponse): Listing[] =>
  rawResponse.results.map(rawResultToListing);
