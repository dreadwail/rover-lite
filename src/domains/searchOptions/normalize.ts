import {
  BasicOption,
  BooleanOption,
  ChoiceOption,
  DateOption,
  IntegerOption,
  MultipleChoiceOption,
  Option,
  Options,
} from '../../state/searchOptions/types';

interface RawBasicOption {
  readonly type: string;
  readonly required: boolean;
  readonly label: string;
  readonly help_text?: string;
  readonly category?: string;
  readonly category_rank?: number;
  readonly category_display_name?: string;
  readonly rank?: number;
}

interface RawDateOption extends RawBasicOption {
  readonly type: 'date';
}

interface RawChoice {
  readonly value: any;
  readonly display_name: string;
}

interface RawChoiceOption extends RawBasicOption {
  readonly type: 'choice';
  readonly choices: RawChoice[];
}

interface RawMultipleChoiceOption extends RawBasicOption {
  readonly type: 'multiple choice';
  readonly choices: RawChoice[];
}

interface RawIntegerOption extends RawBasicOption {
  readonly type: 'integer';
  readonly min_value: number;
  readonly max_value: number;
}

interface RawBooleanOption extends RawBasicOption {
  readonly type: 'boolean';
}

type RawOption =
  | RawBasicOption
  | RawDateOption
  | RawChoiceOption
  | RawMultipleChoiceOption
  | RawIntegerOption
  | RawBooleanOption;

const isRawBooleanOption = (rawOption: RawOption): rawOption is RawBooleanOption =>
  rawOption.type === 'boolean';
const isRawIntegerOption = (rawOption: RawOption): rawOption is RawIntegerOption =>
  rawOption.type === 'integer';
const isRawMultipleChoiceOption = (rawOption: RawOption): rawOption is RawMultipleChoiceOption =>
  rawOption.type === 'multiple choice';
const isRawChoiceOption = (rawOption: RawOption): rawOption is RawChoiceOption =>
  rawOption.type === 'choice';
const isRawDateOption = (rawOption: RawOption): rawOption is RawDateOption =>
  rawOption.type === 'date';

type OptionsResponseActionType = 'GET';

type RawOptions = Record<string, RawOption>;

export interface OptionsResponse {
  readonly name: string;
  readonly description: string;
  readonly renders: string[];
  readonly parses: string[];
  readonly actions: Record<OptionsResponseActionType, RawOptions>;
}

const extractRawCategory = (rawBasicOption: RawBasicOption): Option['category'] => {
  const defaultCategory: Option['category'] = {
    categoryId: 'detail',
    rank: 0,
    displayName: 'Detail',
  };
  if (!rawBasicOption.category) {
    return defaultCategory;
  }
  if (!rawBasicOption.category_display_name) {
    return defaultCategory;
  }
  if (!rawBasicOption.category_rank) {
    return defaultCategory;
  }
  return {
    categoryId: rawBasicOption.category,
    rank: rawBasicOption.category_rank,
    displayName: rawBasicOption.category_display_name,
  };
};

const labelOverrides: Record<string, string> = {
  Minprice: 'Minimum Price',
  Maxprice: 'Maximum Price',
};

const convertOverridableLabel = (rawLabel: string): string => {
  const override = labelOverrides[rawLabel];
  if (override) {
    return override;
  }
  return rawLabel;
};

const rawBasicOptionToOption = (rawBasicOption: RawBasicOption): BasicOption => ({
  type: rawBasicOption.type,
  required: !!rawBasicOption.required,
  label: convertOverridableLabel(rawBasicOption.label),
  helpText: rawBasicOption.help_text,
  rank: rawBasicOption.rank || 0,
  category: extractRawCategory(rawBasicOption),
});

const rawBooleanOptionToOption = (rawBooleanOption: RawBooleanOption): BooleanOption => {
  const option = rawBasicOptionToOption(rawBooleanOption);
  return { ...option, type: 'boolean' };
};

const rawIntegerOptionToOption = (rawIntegerOption: RawIntegerOption): IntegerOption => {
  const option = rawBasicOptionToOption(rawIntegerOption);
  return {
    ...option,
    type: 'integer',
    minValue: rawIntegerOption.min_value,
    maxValue: rawIntegerOption.max_value,
  };
};

const choiceDisplayNameOverrides: Record<string, string> = {
  no_pref: 'No preference',
  no_other_dogs: 'No other dogs',
  other_dogs: 'Other dogs',
  houses: 'Houses',
  apartments: 'Apartments',
};

const convertOverridableChoiceDisplayName = (rawDisplayName: string): string => {
  const override = choiceDisplayNameOverrides[rawDisplayName];
  if (override) {
    return override;
  }
  return rawDisplayName;
};

const rawChoiceOptionToOption = (rawChoiceOption: RawChoiceOption): ChoiceOption => {
  const option = rawBasicOptionToOption(rawChoiceOption);
  return {
    ...option,
    type: 'choice',
    choices: rawChoiceOption.choices.map(choice => ({
      value: choice.value,
      displayName: convertOverridableChoiceDisplayName(choice.display_name),
    })),
  };
};

const rawMultipleChoiceOptionToOption = (
  rawMultipleChoiceOption: RawMultipleChoiceOption
): MultipleChoiceOption => {
  const option = rawBasicOptionToOption(rawMultipleChoiceOption);
  return {
    ...option,
    type: 'multiple choice',
    choices: rawMultipleChoiceOption.choices.map(choice => ({
      value: choice.value,
      displayName: convertOverridableChoiceDisplayName(choice.display_name),
    })),
  };
};

const rawDateOptionToOption = (rawDateOption: RawDateOption): DateOption => {
  const option = rawBasicOptionToOption(rawDateOption);
  return { ...option, type: 'date' };
};

const rawOptionToOption = (rawOption: RawOption): Option => {
  if (isRawBooleanOption(rawOption)) {
    return rawBooleanOptionToOption(rawOption);
  }
  if (isRawIntegerOption(rawOption)) {
    return rawIntegerOptionToOption(rawOption);
  }
  if (isRawMultipleChoiceOption(rawOption)) {
    return rawMultipleChoiceOptionToOption(rawOption);
  }
  if (isRawChoiceOption(rawOption)) {
    return rawChoiceOptionToOption(rawOption);
  }
  if (isRawDateOption(rawOption)) {
    return rawDateOptionToOption(rawOption);
  }
  return rawBasicOptionToOption(rawOption);
};

const keysToSkip: string[] = ['service_type', 'page'];

export const normalize = (response: OptionsResponse): Options => {
  const rawOptions = response.actions.GET;
  return Object.keys(rawOptions).reduce((options, key) => {
    if (keysToSkip.includes(key)) {
      return options;
    }
    const rawOption = rawOptions[key];
    const option = rawOptionToOption(rawOption);
    return { ...options, [key]: option };
  }, {});
};
