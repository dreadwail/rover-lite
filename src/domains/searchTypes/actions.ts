import { Action } from 'redux';

import { request } from '../../api';
import { Endpoints } from '../../state/searchTypes/types';
import { PromiseThunkAction } from '../types';

export interface FetchSearchTypesPendingAction extends Action {
  readonly type: 'FETCH_SEARCH_TYPES.PENDING';
}

export interface FetchSearchTypesFulfilledAction extends Action {
  readonly type: 'FETCH_SEARCH_TYPES.FULFILLED';
  readonly payload: Endpoints;
}

export interface FetchSearchTypesRejectedAction extends Action {
  readonly type: 'FETCH_SEARCH_TYPES.REJECTED';
  readonly payload: {
    readonly error: any;
  };
  readonly error: true;
}

const fetchSearchTypesPending = (): FetchSearchTypesPendingAction => ({
  type: 'FETCH_SEARCH_TYPES.PENDING',
});

const fetchSearchTypesFulfilled = (endpoints: Endpoints): FetchSearchTypesFulfilledAction => ({
  type: 'FETCH_SEARCH_TYPES.FULFILLED',
  payload: endpoints,
});

const fetchSearchTypesRejected = (error: any): FetchSearchTypesRejectedAction => ({
  type: 'FETCH_SEARCH_TYPES.REJECTED',
  payload: {
    error,
  },
  error: true,
});

export type FetchSearchTypes = () => PromiseThunkAction<void>;
export const fetchSearchTypes: FetchSearchTypes = () => async dispatch => {
  dispatch(fetchSearchTypesPending());
  try {
    const response = await request<Endpoints>(
      'GET',
      'https://www.rover.com/api/v3/search-endpoints/'
    );
    dispatch(fetchSearchTypesFulfilled(response));
  } catch (err) {
    dispatch(fetchSearchTypesRejected(err));
  }
};
