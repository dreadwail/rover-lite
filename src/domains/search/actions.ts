import { Action } from 'redux';

import { SearchType } from '../../state/searchTypes/types';
import { GenericThunkAction } from '../types';

import { getInputs } from './selectors';

export interface SearchTypeChangedAction extends Action {
  readonly type: 'SEARCH.TYPE_CHANGED';
  readonly payload: SearchType;
}

export interface SearchFieldChangedAction extends Action {
  readonly type: 'SEARCH.FIELD_CHANGED';
  readonly payload: {
    readonly key: string;
    readonly value: any;
  };
}

export const searchTypeChanged = (searchType: SearchType): SearchTypeChangedAction => ({
  type: 'SEARCH.TYPE_CHANGED',
  payload: searchType,
});

export type SearchFieldChanged = (key: string, value: any) => GenericThunkAction<void>;
export const searchFieldChanged: SearchFieldChanged = (key, value) => async (
  dispatch,
  getState
) => {
  const state = getState();
  const inputs = getInputs(state);
  const startDate = inputs.start_date;
  const endDate = inputs.end_date;
  if (startDate != null && key === 'end_date' && startDate > value) {
    return;
  }
  if (endDate != null && key === 'start_date' && value > endDate) {
    return;
  }
  const action: SearchFieldChangedAction = {
    type: 'SEARCH.FIELD_CHANGED',
    payload: { key, value },
  };
  dispatch(action);
};
