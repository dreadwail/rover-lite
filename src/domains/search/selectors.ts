import { SearchInputs } from '../../state/search/types';
import { SearchType } from '../../state/searchTypes/types';
import { RoverSearchState } from '../../state/types';

export const getSearchType = (state: RoverSearchState): SearchType => state.search.searchType;
export const getInputs = (state: RoverSearchState): SearchInputs => state.search.inputs;
export const getLatLong = (state: RoverSearchState): [number, number] => [
  state.search.latitude,
  state.search.longitude,
];
