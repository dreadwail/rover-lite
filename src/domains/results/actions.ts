import { Action } from 'redux';

import { request } from '../../api/http';
import { urlEncodeQuery } from '../../api/query';
import { Listing } from '../../state/results/types';
import { SearchInputs } from '../../state/search/types';
import debouncedAction from '../debouncedThunk';
import { getInputs, getLatLong, getSearchType } from '../search/selectors';
import { PromiseThunkAction } from '../types';

import { normalize, RawSearchResponse } from './normalize';

export interface FetchResultsPendingAction extends Action {
  readonly type: 'FETCH_RESULTS.PENDING';
}

export interface FetchResultsFulfilledAction extends Action {
  readonly type: 'FETCH_RESULTS.FULFILLED';
  readonly payload: Listing[];
}

export interface FetchResultsRejectedAction extends Action {
  readonly type: 'FETCH_RESULTS.REJECTED';
  readonly payload: {
    readonly error: any;
  };
  readonly error: true;
}

const fetchResultsPending = (): FetchResultsPendingAction => ({
  type: 'FETCH_RESULTS.PENDING',
});

const fetchResultsFulfilled = (listings: Listing[]): FetchResultsFulfilledAction => ({
  type: 'FETCH_RESULTS.FULFILLED',
  payload: listings,
});

const fetchResultsRejected = (error: any): FetchResultsRejectedAction => ({
  type: 'FETCH_RESULTS.REJECTED',
  payload: {
    error,
  },
  error: true,
});

export type FetchResults = () => PromiseThunkAction<void>;
export const fetchResults: FetchResults = () => async (dispatch, getState) => {
  dispatch(fetchResultsPending());
  try {
    const state = getState();

    const searchType = getSearchType(state);
    const [latitude, longitude] = getLatLong(state);

    const defaultInputs: SearchInputs = {
      centerlat: latitude,
      centerlng: longitude,
      service_type: searchType,
    };
    const userInputs = getInputs(state);
    const inputs = { ...defaultInputs, ...userInputs };
    const encodedQuery = urlEncodeQuery(inputs);

    const urlBase = 'https://www.rover.com/api/v3/search';
    const url = `${urlBase}/${searchType}/?${encodedQuery}`;

    const response = await request<RawSearchResponse>('GET', url);

    const normalizedListings = normalize(response);
    dispatch(fetchResultsFulfilled(normalizedListings));
  } catch (err) {
    dispatch(fetchResultsRejected(err));
  }
};

export const debouncedFetchResults = debouncedAction(fetchResults, 250);
