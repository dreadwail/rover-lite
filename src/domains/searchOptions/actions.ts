import { Action } from 'redux';

import { request } from '../../api';
import { Options } from '../../state/searchOptions/types';
import { SearchType } from '../../state/searchTypes/types';
import { getEndpointUrl } from '../searchTypes/selectors';
import { PromiseThunkAction } from '../types';

import { normalize, OptionsResponse } from './normalize';

export interface FetchSearchOptionsPendingAction extends Action {
  readonly type: 'FETCH_SEARCH_OPTIONS.PENDING';
  readonly payload: {
    readonly searchType: SearchType;
  };
}

export interface FetchSearchOptionsFulfilledAction extends Action {
  readonly type: 'FETCH_SEARCH_OPTIONS.FULFILLED';
  readonly payload: {
    readonly searchType: SearchType;
    readonly options: Options;
  };
}

export interface FetchSearchOptionsRejectedAction extends Action {
  readonly type: 'FETCH_SEARCH_OPTIONS.REJECTED';
  readonly payload: {
    readonly searchType: SearchType;
    readonly error: any;
  };
  readonly error: true;
}

const fetchSearchOptionsPending = (searchType: SearchType): FetchSearchOptionsPendingAction => ({
  type: 'FETCH_SEARCH_OPTIONS.PENDING',
  payload: {
    searchType,
  },
});

const fetchSearchOptionsFulfilled = (
  searchType: SearchType,
  options: Options
): FetchSearchOptionsFulfilledAction => ({
  type: 'FETCH_SEARCH_OPTIONS.FULFILLED',
  payload: {
    searchType,
    options,
  },
});

const fetchSearchOptionsRejected = (
  searchType: SearchType,
  error: any
): FetchSearchOptionsRejectedAction => ({
  type: 'FETCH_SEARCH_OPTIONS.REJECTED',
  payload: {
    searchType,
    error,
  },
  error: true,
});

export type FetchSearchOptions = (searchType: SearchType) => PromiseThunkAction<void>;
export const fetchSearchOptions: FetchSearchOptions = searchType => async (dispatch, getState) => {
  dispatch(fetchSearchOptionsPending(searchType));
  try {
    const state = getState();
    const endpointUrl = getEndpointUrl(state, searchType);
    if (!endpointUrl) {
      throw new Error(`Unhandled search type: ${searchType}`);
    }
    const response = await request<OptionsResponse>('OPTIONS', endpointUrl);
    const normalizedOptions = normalize(response);
    dispatch(fetchSearchOptionsFulfilled(searchType, normalizedOptions));
  } catch (err) {
    dispatch(fetchSearchOptionsRejected(searchType, err));
  }
};
