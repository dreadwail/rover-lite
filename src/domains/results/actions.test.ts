jest.mock('../../api/http');

import { request } from '../../api/http';
import { createMockState } from '../../state/mockState';
import { createMockStore, MockStore } from '../../state/mockStore';
import { Search } from '../../state/search/types';
import { RoverSearchState } from '../../state/types';

import { fetchResults } from './actions';
import { RawResult, RawSearchResponse } from './normalize';

describe('results actions', () => {
  let state: RoverSearchState;
  let store: MockStore<RoverSearchState>;

  beforeEach(() => {
    const search: Search = {
      searchType: 'overnight-boarding',
      latitude: 47.6843725,
      longitude: -122.3874837,
      inputs: {},
    };
    state = createMockState({ search });
    store = createMockStore<RoverSearchState>(state);
  });

  describe('.fetchResults', () => {
    describe('when the api call fails', () => {
      const error = new Error('bang');

      beforeEach(() => {
        (request as jest.Mock).mockImplementation(() => Promise.reject(error));
        return store.dispatch(fetchResults());
      });

      it('calls the right api', () => {
        const { latitude, longitude, searchType } = state.search;
        const urlBase = 'https://www.rover.com/api/v3/search';
        const query = `centerlat=${latitude}&centerlng=${longitude}&service_type=${searchType}`;
        const expectedUrl = `${urlBase}/${searchType}/?${query}`;
        expect(request).toHaveBeenCalledWith('GET', expectedUrl);
      });

      it('dispatches the correct actions', () => {
        expect(store.getActions()).toMatchSnapshot();
      });
    });

    describe('when the api call succeeds', () => {
      beforeEach(() => {
        const results: RawResult[] = [
          {
            rank: 0,
            person_summary: {
              responsiveness: {
                photo_rate_desc: 'a',
                percent_desc: 'b',
                time_short_desc: 'c',
              },
              first_name: 'd',
              reviews_count: 4,
              profile_photo: {
                small: 'e',
              },
              ratings_average: '4.8',
              web_url: 'f',
              images: [],
              provider_profile: {
                hosting_service_preferences: {
                  dogs_preferences_attributes: [],
                  accepted_dogs_attributes: [],
                  dogs_experience_attributes: [],
                  attributes: [],
                },
                traveling_service_preferences: {
                  dogs_preferences_attributes: [],
                  accepted_dogs_attributes: [],
                  dogs_experience_attributes: [],
                  attributes: [],
                },
                attributes: [],
                years_of_experience: 4,
              },
              attributes: {},
              badges: [],
            },
            reviews_count: 2,
            neighborhood: 'Ballard',
            price_unit: 'night',
            city: 'Seattle',
            zip: '98117',
            title: 'g',
            state: 'WA',
            contact_url: 'h',
            person_name: 'i',
            price: '42.0',
            review_text: 'j',
            review_dog_photo_url: 'k',
            badge_data: [],
            repeat_client_count: 5,
            reviews: [],
            ratings_average: '4.8',
            currency_symbol: '$',
          },
        ];
        const response: RawSearchResponse = { results };
        (request as jest.Mock).mockImplementation(() => Promise.resolve(response));
        return store.dispatch(fetchResults());
      });

      it('calls the right api', () => {
        const { latitude, longitude, searchType } = state.search;
        const urlBase = 'https://www.rover.com/api/v3/search';
        const query = `centerlat=${latitude}&centerlng=${longitude}&service_type=${searchType}`;
        const expectedUrl = `${urlBase}/${searchType}/?${query}`;
        expect(request).toHaveBeenCalledWith('GET', expectedUrl);
      });

      it('dispatches the correct actions', () => {
        expect(store.getActions()).toMatchSnapshot();
      });
    });
  });
});
