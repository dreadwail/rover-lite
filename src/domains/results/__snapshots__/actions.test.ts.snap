// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`results actions .fetchResults when the api call fails dispatches the correct actions 1`] = `
Array [
  Object {
    "type": "FETCH_RESULTS.PENDING",
  },
  Object {
    "error": true,
    "payload": Object {
      "error": [Error: bang],
    },
    "type": "FETCH_RESULTS.REJECTED",
  },
]
`;

exports[`results actions .fetchResults when the api call succeeds dispatches the correct actions 1`] = `
Array [
  Object {
    "type": "FETCH_RESULTS.PENDING",
  },
  Object {
    "payload": Array [
      Object {
        "badges": Array [],
        "city": "Seattle",
        "currencySymbol": "$",
        "featuredReview": "j",
        "name": "d",
        "neighborhood": "Ballard",
        "price": 42,
        "priceUnit": "night",
        "rank": 0,
        "ratingAverage": 4.8,
        "repeatClientCount": 5,
        "reviewCount": 2,
        "state": "WA",
        "tagline": "g",
        "thumbnailUrl": "e",
        "webUrl": "f",
        "zip": "98117",
      },
    ],
    "type": "FETCH_RESULTS.FULFILLED",
  },
]
`;
